tools/functional-tester/etcd-tester/stresser.go

```diff
@@ -37,6 +38,96 @@ func init() {
 	grpclog.SetLogger(plog)
 }
 
+type stressFunc func(ctx context.Context, kvc pb.KVClient, key, val []byte) error
+
+func stressFuncPut(ctx context.Context, kvc pb.KVClient, key, val []byte) error {
+	_, err := kvc.Put(ctx, &pb.PutRequest{Key: key, Value: val}, grpc.FailFast(false))
+	return err
+}
+
+func stressFuncRange(ctx context.Context, kvc pb.KVClient, key, val []byte) error {
+	_, err := kvc.Range(ctx, &pb.RangeRequest{
+		Key: key,
+	}, grpc.FailFast(false))
+	return err
+}
+
+func stressFuncRangePrefix(ctx context.Context, kvc pb.KVClient, key, val []byte) error {
+	_, err := kvc.Range(ctx, &pb.RangeRequest{
+		Key:      []byte("foo"),
+		RangeEnd: []byte(fmt.Sprintf("foo%d", rand.Intn(250000))),
+	}, grpc.FailFast(false))
+	return err
+}
+
+func stressFuncRangeAll(ctx context.Context, kvc pb.KVClient, key, val []byte) error {
+	_, err := kvc.Range(ctx, &pb.RangeRequest{
+		Key:      []byte("\x00"),
+		RangeEnd: []byte("\x00"),
+	}, grpc.FailFast(false))
+	return err
+}
+
+func stressFuncDeleteRandom(ctx context.Context, kvc pb.KVClient, key, val []byte) error {
+	_, err := kvc.DeleteRange(ctx, &pb.DeleteRangeRequest{
+		Key: []byte(fmt.Sprintf("foo%d", rand.Intn(250000))),
+	}, grpc.FailFast(false))
+	return err
+}
+
+func stressFuncDeletePrefix(ctx context.Context, kvc pb.KVClient, key, val []byte) error {
+	_, err := kvc.DeleteRange(ctx, &pb.DeleteRangeRequest{
+		Key:      []byte("foo"),
+		RangeEnd: []byte(fmt.Sprintf("foo%d", rand.Intn(250000))),
+	}, grpc.FailFast(false))
+	return err
+}
+
+type stressEntry struct {
+	weight float32
+	f      stressFunc
+}
+
+var stressEntries = []stressEntry{
+	{weight: 0.5, f: stressFuncPut},
+	{weight: 0.1, f: stressFuncRange},
+	{weight: 0.1, f: stressFuncRangePrefix},
+	{weight: 0.1, f: stressFuncRangeAll},
+	{weight: 0.1, f: stressFuncDeleteRandom},
+	{weight: 0.1, f: stressFuncDeletePrefix},
+}
+
+type stressTable struct {
+	entries       []stressEntry
+	distributions []float32
+}
+
+func (wt stressTable) Len() int           { return len(wt.entries) }
+func (wt stressTable) Swap(i, j int)      { wt.entries[i], wt.entries[j] = wt.entries[j], wt.entries[i] }
+func (wt stressTable) Less(i, j int) bool { return wt.entries[i].weight < wt.entries[j].weight }
+
+func createStressTable(entries []stressEntry) *stressTable {
+	wt := stressTable{entries: entries}
+	sort.Sort(wt)
+	var cw float32
+	for _, entry := range wt.entries {
+		cw += entry.weight
+		wt.distributions = append(wt.distributions, cw)
+	}
+	return &wt
+}
+
+func (wt *stressTable) choose() stressFunc {
+	entryN := len(wt.entries)
+	lastWeight := wt.entries[len(wt.entries)-1].weight
+	idx := sort.Search(entryN, func(i int) bool {
```

Comment:
Sorting/searching seems like overkill since the table is relatively small (also tougher to get right). A simpler algorithm:
```go
v := rand.Float64() * wt.sumWeights
sum := float64(0)
for i := range wt.entries {
    sum += wt.entries[i].weight
    if sum >= v {
        return wt.entries[i].f
    }
}
```